apps/web/src/pages/settings/tabs/EmailSettings.tsx

```diff
@@ -0,0 +1,175 @@
+import { useClipboard } from '@mantine/hooks';
+import styled from '@emotion/styled';
+import { inputStyles } from '../../../design-system/config/inputs.styles';
+import Card from '../../../components/layout/components/Card';
+import { ActionIcon, Center, Input as MantineInput } from '@mantine/core';
+import { colors, Text, Input, Tooltip, Button } from '../../../design-system';
+import { Check, SuccessIcon, Copy } from '../../../design-system/icons';
+import React, { useEffect } from 'react';
+import { Control, Controller, useForm } from 'react-hook-form';
+import { useEnvController } from '../../../store/use-env-controller';
+import { useMutation } from '@tanstack/react-query';
+import { updateDnsSettings, validateMxRecord } from '../../../api/environment';
+import { showNotification } from '@mantine/notifications';
+import { WarningIcon } from '../../../design-system/icons/statuses/WarningIcon';
+
+export const EmailSettings = () => {
+  const clipboardEnvironmentIdentifier = useClipboard({ timeout: 1000 });
+  const MAIL_SERVER_DOMAIN = '10 postal-dev.novu.co';
+  const { readonly, environment, refetchEnvironment } = useEnvController();
+
+  const { mutateAsync: updateDnsSettingsMutation, isLoading: isUpdateDnsSettingsLoading } = useMutation<
+    { dns: { mxRecordConfigured: string; domain: string } },
+    { error: string; message: string; statusCode: number },
+    { payload: { domain: string | undefined }; environmentId: string }
+  >(({ payload, environmentId }) => updateDnsSettings(payload, environmentId));
+
+  const { setValue, handleSubmit, control } = useForm({
+    defaultValues: {
+      mxRecordConfigured: false,
+      domain: '',
+    },
+  });
+
+  useEffect(() => {
+    if (environment) {
+      if (environment.dns?.domain) {
+        setValue('domain', environment.dns?.domain);
+      }
+      if (environment.dns?.mxRecordConfigured) {
+        setValue('mxRecordConfigured', environment.dns?.mxRecordConfigured);
+      }
+    }
+  }, [environment]);
+
+  useEffect(() => {
+    handleCheckRecords();
+  }, []);
+
+  async function handleUpdateDnsSettings({ domain }) {
+    const payload = {
+      domain,
+    };
+
+    await updateDnsSettings(payload, environment?._id ?? '');
+
+    showNotification({
+      message: 'Domain info updated successfully',
+      color: 'green',
+    });
+  }
+
+  async function handleCheckRecords() {
+    const record = await validateMxRecord();
+
+    // eslint-disable-next-line no-console
+    console.log('record ', record);
+
+    if (record.mxRecordConfigured !== environment?.dns?.mxRecordConfigured) {
+      await refetchEnvironment();
+    }
+  }
+
+  return (
```

Comment:
Let's make sure that all UI things relevant to this, are not visible on the FrontEnd if self hosted